R/utils.R

```diff
@@ -58,8 +58,16 @@ include_parent_datanames <- function(dataname, join_keys) {
 #' @keywords internal
 teal_data_to_filtered_data <- function(x, datanames = teal.data::datanames(x)) {
   checkmate::assert_class(x, "teal_data")
-  checkmate::assert_character(datanames, min.len = 1L, any.missing = FALSE)
-  checkmate::assert_subset(datanames, teal.data::datanames(x))
+  checkmate::assert_character(datanames, any.missing = FALSE)
+
+  if (length(datanames) == 0) {
+    datanames <- ls(x@env)
```

Comment:
Should we add an abstraction layer here instead of directly accessing the slot, neither `teal_data` nor `qenv` objects are defined in {teal}.

If this makes sense, then we could:
- register a S4 method for `teal.code::get_var("qenv", "missing")`
- register a new S4 method `teal.code::get_vars("qenv")`
- apply this only to `teal.data` with a new method `teal.data::get_vars("teal_data")`

Reply:
there is feature request for this actually https://github.com/insightsengineering/teal.code/issues/175

Reply:
this will be modified once that is implemented

Reply:
There is an issue for that somewhere.